pkg/sdk/random.go

```diff
@@ -1,77 +1,43 @@
 package sdk
 
 import (
-	"testing"
-
-	"github.com/brianvoe/gofakeit/v6"
-	"github.com/hashicorp/go-uuid"
-	"github.com/stretchr/testify/require"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk/internal/random"
 )
 
-func randomUUID(t *testing.T) string {
-	t.Helper()
-	v, err := uuid.GenerateUUID()
-	require.NoError(t, err)
-	return v
-}
-
-func randomComment(t *testing.T) string {
-	t.Helper()
-	return gofakeit.Sentence(10)
-}
-
-func randomBool(t *testing.T) bool {
-	t.Helper()
-	return gofakeit.Bool()
-}
-
-func randomString(t *testing.T) string {
-	t.Helper()
-	return gofakeit.Password(true, true, true, true, false, 28)
-}
-
-func randomStringN(t *testing.T, num int) string {
-	t.Helper()
-	return gofakeit.Password(true, true, true, true, false, num)
-}
-
-func randomAlphanumericN(t *testing.T, num int) string {
-	t.Helper()
-	return gofakeit.Password(true, true, true, false, false, num)
-}
-
-func randomStringRange(t *testing.T, min, max int) string {
-	t.Helper()
-	if min > max {
-		t.Errorf("min %d is greater than max %d", min, max)
-	}
-	return gofakeit.Password(true, true, true, true, false, randomIntRange(t, min, max))
-}
-
-func randomIntRange(t *testing.T, min, max int) int {
-	t.Helper()
-	if min > max {
-		t.Errorf("min %d is greater than max %d", min, max)
-	}
-	return gofakeit.IntRange(min, max)
-}
-
-func randomSchemaObjectIdentifier(t *testing.T) SchemaObjectIdentifier {
-	t.Helper()
-	return NewSchemaObjectIdentifier(randomStringN(t, 12), randomStringN(t, 12), randomStringN(t, 12))
-}
-
-func randomDatabaseObjectIdentifier(t *testing.T) DatabaseObjectIdentifier {
-	t.Helper()
-	return NewDatabaseObjectIdentifier(randomStringN(t, 12), randomStringN(t, 12))
-}
-
-func alphanumericDatabaseObjectIdentifier(t *testing.T) DatabaseObjectIdentifier {
-	t.Helper()
-	return NewDatabaseObjectIdentifier(randomAlphanumericN(t, 12), randomAlphanumericN(t, 12))
-}
-
-func randomAccountObjectIdentifier(t *testing.T) AccountObjectIdentifier {
-	t.Helper()
-	return NewAccountObjectIdentifier(randomStringN(t, 12))
+// Helper methods in this file are used both in SDK tests and also in integration tests.
+// These methods are not supposed to be used in production code, so it would be better to not export them.
+// Simply moving them to internal/random along other random helper methods is not an easy option, because it would create import cycle:
+// sdk needed here -> sdk imports this to use the methods for tests.
```

Comment:
maybe moving unit tests into test package could be an option ?

Reply:
Didn't read the solutions below, but after reading them, separate package for unit tests seems the best solution (imo). Don't know if separate package or sdk_test would be better, but the overall idea of extracting unit tests is pretty good + solution to avoid circular deps.

Reply:
Thanks for the response. I like the most the option with extracting identifiers to separate package inside SDK package.

The option with separate package for tests is also fine, if we agree to change the approach in unit tests for black-box one.